src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -229,29 +232,35 @@ public function handle(GetOrderForViewing $query): OrderForViewing
      *
      * @return OrderCustomerForViewing|null
      */
-    private function getOrderCustomer(Order $order): ?OrderCustomerForViewing
+    private function getOrderCustomer(Order $order, OrderInvoiceAddressForViewing $invoiceAddress): ?OrderCustomerForViewing
     {
         $currency = new Currency($order->id_currency);
         $customer = new Customer($order->id_customer);
+        $genderName = '';
+        $totalSpentSinceRegistration = null;
 
         if (!Validate::isLoadedObject($customer)) {
-            return null;
-        }
+            $cart = new Cart($order->id_cart);
```

Comment:
I think that if a developer reads this code in a few month, he might be thinking "what? when there is no customer in the order, they create it? what is this logic? I don't understand"

So I suggest we provide the context to this reader
- either by adding a comment
```suggestion
            // if there is no valid customer attached to the order, the customer must have been deleted
            // from the database. we then create a fake customer object, using the invoice address data
            $cart = new Cart($order->id_cart);
```

We could also put this explanation inside a `private` method name
```suggestion
            $customer->buildFakeCustomerObjectFromAddressData($invoiceAddress);
```

Reply:
Thanks for recommendation, I update my PR